templates/chatbot/template.yaml

```diff
@@ -346,12 +350,12 @@ spec:
           # for vllm
           vllmSelected: ${{ parameters.modelServer === 'vLLM' }}
           vllmModelServiceContainer: quay.io/rh-aiservices-bu/vllm-openai-ubi9:0.4.2
-          modelName: ${{ parameters.modelName if parameters.modelServer === 'Existing model server' else 'instructlab/granite-7b-lab' }}
+          modelName: ${{ parameters.modelName | parseEntityRef | pick('name') if parameters.modelServer === 'Existing model server' else 'instructlab/granite-7b-lab' }}
```

Comment:
This plucks just the model name from the Backstage Entity URL, i.e just the `granite-31-8b-lab-v1-140-version-1` portion from `component:default/granite-31-8b-lab-v1-140-version-1`

Reply:
Works for me. I have deployed a model in a test cluster through RHOAI and I was able to fetch (autimatically) the model name. Once the templated application was up, I can verify that the `MODEL_NAME` and `MODEL_ENDPOINT` passed correctly to my application.

![Screenshot from 2025-05-02 15-55-59](https://github.com/user-attachments/assets/51950a64-a65f-4c06-b6c0-a247a4cb46dd)